tests/Core.nut

```diff
@@ -0,0 +1,422 @@
+//    If more than one requests send, then test will be passed only when all responses verifications passed.
+// 
+// @param {table} params
+// @param {string} expect - Expected completion of the test (success|fail)
+// @return {Promise}
+function createTest(params = {}, expect = "success") {
+    return Promise(function(resolve, reject) {
+        local success = function() {
+            if (expect == "fail") {
+                reject("createTest test was resolved, but 'fail' expected");
+            } else {
+                resolve();
+            }
+        };
+        local fail = function(reason = null) {
+            if (expect == "fail") {
+                resolve();
+            } else {
+                reject(reason);
+            }
+        };
+        local app;
+        local route;
+        // Setup default values
+        params.setdelegate(defaultParams);
+        try {
+            // Call Rocky.Route constructor directly
+            if (params.paramsRockyRoute != null) {
+                local illegalRoute = Rocky.Route(params.paramsRockyRoute);
+                if (params.onAuthorizeRoute != null) {
+                    illegalRoute.authorize(params.onAuthorizeRoute);
+                }
+                if (params.onUnauthorizedRoute != null) {
+                    illegalRoute.onUnauthorized(params.onUnauthorizedRoute);
+                }
+                if (params.onTimeoutRoute != null) {
+                    illegalRoute.onTimeout(params.onTimeoutRoute);
+                }
+                if (params.onExceptionRoute != null) {
+                    illegalRoute.onException(params.onExceptionRoute);
+                }
+                illegalRoute.use(params.mw);
+            }
+            // Call Rocky.Context constructor directly
+            if (params.paramsRockyContext != null) {
+                local illegalContext = Rocky.Context(params.paramsRockyContext[0], params.paramsRockyContext[1]);
+                if (params.paramsRockyContextAdditionalUsage != null && params.paramsRockyContextAdditionalUsage) {
+                    local tmp;
+                    tmp = illegalContext.isComplete();
+                    illegalContext.setHeader("Header", "here");
+                    tmp = illegalContext.getHeader("Header");
+                    tmp = illegalContext.req.method;
+                    tmp = illegalContext.req.path;
+                    tmp = illegalContext.req.query;
+                    tmp = illegalContext.req.headers;
+                    tmp = illegalContext.req.body;
+                    tmp = illegalContext.id;
+                    tmp = illegalContext.userdata;
+                    tmp = illegalContext.path;
+                    tmp = illegalContext.matches;
+                    tmp = illegalContext.isbrowser();
+                    illegalContext.send(200);
+                }
+            }
+        } catch(ex) {
+            fail("Unexpected error while call constructors directly: " + ex);

```

Comment:
The fail function does not imply a failure of the test. The failure or success of a test depends on the 'expect' parameter of the createTest function. Tests aimed at calling the Context/Route constructor expect that an error will occur.